cpu/samd5x/cpu.c

```diff
@@ -197,24 +201,34 @@ static void fdpll_init_nolock(uint8_t idx, uint32_t f_cpu, uint8_t flags)
         return;
     }
 
-    /* Source the DPLL from 32kHz GCLK1 ( equivalent to ((f_cpu << 5) / 32768) ) */
-    const uint32_t LDR = (f_cpu >> 10);
-
     /* disable the DPLL before changing the configuration */
     OSCCTRL->Dpll[idx].DPLLCTRLA.reg &= ~OSCCTRL_DPLLCTRLA_ENABLE;
     while (OSCCTRL->Dpll[idx].DPLLSYNCBUSY.reg) {}
 
-    /* set DPLL clock source */
-    GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL0 + idx].reg = GCLK_PCHCTRL_GEN(1) | GCLK_PCHCTRL_CHEN;
-    while (!(GCLK->PCHCTRL[OSCCTRL_GCLK_ID_FDPLL0 + idx].reg & GCLK_PCHCTRL_CHEN)) {}
-
-    OSCCTRL->Dpll[idx].DPLLRATIO.reg = OSCCTRL_DPLLRATIO_LDRFRAC(LDR & 0x1F)
-                                     | OSCCTRL_DPLLRATIO_LDR((LDR >> 5) - 1);
-
-    /* Without LBYPASS, startup takes very long, see errata section 2.13. */
-    OSCCTRL->Dpll[idx].DPLLCTRLB.reg = OSCCTRL_DPLLCTRLB_REFCLK_GCLK
-                                     | OSCCTRL_DPLLCTRLB_WUF
-                                     | OSCCTRL_DPLLCTRLB_LBYPASS;
+    if(!EXTERNAL_OSC32_SOURCE){
```

Comment:
```suggestion
    if (EXTERNAL_OSC32_SOURCE) {
```
If you re-order the branches you don't need the negation, makes it easier to read.

Reply:
right